### ISSUE TITLE
mboost: add support for various families

### ISSUE THREAD

**Issue by giuseppec** [mlr-org/mlr#662]
Here is a list of families that can be used in mboost: http://www.inside-r.org/packages/cran/mboost/docs/Family
Currently, we only support `Binomial()` and `AdaExp()` for classification (but don't support `Binomial(link = "probit")`) or families that have an additional arg. Also we should think of a solution how to handle additional args in mboost families (as they can be hyperparameters as well, as in the `Huber` family).
For regression we should at least also support `Poisson()` which we currently don't support. 

``` r
library(mlr)
library(mboost)

# works
makeLearner("classif.glmboost", family = mboost::Binomial())
# does not work
makeLearner("classif.glmboost", family = mboost::Binomial(link = "probit"))

# furthermore we do not support poisson regression
makeLearner("regr.glmboost", family = Poisson())
```

**Comment by berndbischl** [comment]
I guess we add the families with hyperpars in the following way:
We set a default for their hyperpars.

We also introduce some a new param family2 (or whatever) which is untypped and you can pass what you want.

ok?

could you do that?

**Comment by berndbischl** [comment]
introduce family.custom as an untyped param. if this is set, it has preference over "family"

**Comment by MariaErdmann** [comment]
These are the families supported by boost_family and some of my thoughts with an arrow behind:
AdaExp()  --> already implemented in the classif-learners
AUC()  --> not implemented yet. Shall we implement this one for mboost's classif learners?
Binomial(link = c("logit", "probit"), ...)  --> right now we have Binomial() in classif_glmboost meaning that the loss function is the negative binomial log-likelihood
Gaussian()  --> default for regression
Huber(d = NULL)  --> for continous response. added this one in PR #1023.
Laplace()  --> for regression. already implemented in regr_glmboost
Poisson()  -->  We talked about adding this into the regression learner (see PR #1023 )
GammaReg(nuirange = c(0, 100))  --> for non-negative responses. I added this one in PR #1023 
CoxPH()  --> for survivial. alread implemented
QuantReg(tau = 0.5, qoffset = 0.5)  --> for regression. I added this one in #1023 
ExpectReg(tau = 0.5)  --> for regression. I added this one in #1023 
NBinomial(nuirange = c(0, 100))
 PropOdds(nuirange = c(-0.5, -1), offrange = c(-5, 5)) 
Weibull(nuirange = c(0, 100))  --> for survival
Loglog(nuirange = c(0, 100)) 
Lognormal(nuirange = c(0, 100))
 Gehan() 
Hurdle(nuirange = c(0, 100))
Multinomial()

**Comment by catastrophic-failure** [comment]
Regarding quantile regression, some quick discussion is due in #1022.

**Comment by MariaErdmann** [comment]
I am not sure if we want to support Multinomial() as a family for classif.glmboost. Because this family requires that the data is converted to a list and the response needs to be defined as a dummy vector with one level less than the outcome which is used as reference category. Do we want to implement  that?
